### PR TITLE
added testdata for crop #24384

### PR DESCRIPTION
**Pull request by Dhanwanth1803**
@dkurt Sorry for closing the previous PR. Here are the changes that you asked for. Thank you.

### CODE REVIEW

File: testdata/dnn/darknet/generate_darknet_models.py [hunk 1]
```diff
@@ -37,4 +37,4 @@ def genTestData(name, inpShape, outShape):
 genTestData('route_multi', [1, 6, 3, 6], [1, 4, 2, 3])
 genTestData('relu', [1, 2, 3, 4], [1, 2, 3, 4])
 genTestData('sam', [1, 2, 3, 4], [1, 4, 3, 4])
-
+genTestData('crop', [1, 3, 5, 6], [1, 3, 3, 3])
```
Comment (dkurt) [inline testdata/dnn/darknet/generate_darknet_models.py:40]: `crop_width` and `crop_height` are output dimensions. So should be `genTestData('crop', [1, 3, 5, 6], [1, 3, 3, 2])`

Comment (Dhanwanth1803) [reply testdata/dnn/darknet/generate_darknet_models.py:40]: You want me to generate files for `genTestData('crop', [1, 3, 5, 6], [1, 3, 3, 2])`?

Comment (dkurt) [reply testdata/dnn/darknet/generate_darknet_models.py:40]: Yes, because correct output size is 1x3x3x2

Comment (Dhanwanth1803) [reply testdata/dnn/darknet/generate_darknet_models.py:40]: Okay. I will do it right away.

**Comment by Dhanwanth1803** [comment]
@dkurt Can you review my latest commit. I generated it for `genTestData('crop', [1, 3, 5, 6], [1, 3, 3, 2])` . Thank you.

**Comment by dkurt** [comment]
@Dhanwanth1803, please create a branch with a name `feat-crop` from `feat-crop1` again and reopen PR. Make sure commit https://github.com/opencv/opencv_extra/pull/1112/commits/198f2dde34f5c007be0f5ace708da42e9c1f3fb0 is pulled.

**Pull request merged**